module/VuFind/src/VuFind/Config/Feature/ConfigSettingPropertiesTrait.php

```
@@ -0,0 +1,252 @@
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Config
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Config\Feature;
+
+use VuFind\Exception\BadConfig;
+
+use function array_key_exists;
+use function in_array;
+use function is_array;
+
+/**
+ * Trait implementing SettingPropertiesInterface
+ *
+ * @category VuFind
+ * @package  Config
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+trait ConfigSettingPropertiesTrait
+{
+    /**
+     * Required and conditionally required settings.
+     *
+     * @var array
+     */
+    protected array $requiredSettings = [self::DEFAULT_CONTEXT => []];
+
+    /**
+     * Localizable settings.
+     *
+     * @var array
+     */
+    protected array $localizableSettings = [self::DEFAULT_CONTEXT => []];
+
+    /**
+     * Return required and conditionally required settings.
+     *
+     * @param string $contextKey Key identifying the context (optional)
+     *
+     * @return array
+     */
+    public function getRequiredSettings(
+        string $contextKey = self::DEFAULT_CONTEXT
+    ): array {
+        return $this->requiredSettings[$contextKey];
```

This will trigger a warning if `$contextKey` is not a defined value. Should we `?? []` or throw an exception when we encounter an unexpected context?